tests/phpunit/CRM/MembershipExtras/Job/OfflineAutoRenewal/MultiInstallmentPlanTest.php

```diff
@@ -761,4 +763,98 @@ private function getCustomFieldId($customGroupName, $customFieldName) {
     ]);
   }
 
+  public function testRenewalWithNonRenewableLineOnCurrentPeriodAndNewMembershipForNextPeriod() {
```

Comment:
I would also add the following tests : 

1- Two memberships in the current period where non of them is renewal and each have different end date, with one renewal membership in the next tab with.

2- Two memberships in the current period where one of them is renewal and the other is not and each with  diffrenet end date, and another renewal membership in the next tab (which one membership will be renewed and new one with new type will be created).

also, similar tests for single instalment job should be added.